exporter/datadogexporter/traces_exporter_test.go

```diff
@@ -444,10 +444,23 @@ func testPushTraceDataNewEnvConvention(t *testing.T, enableReceiveResourceSpansV
 	assert.Equal(t, "new_env", traces.TracerPayloads[0].GetEnv())
 }
 
-func TestPushTraceData_OperationAndResourceNameV2(t *testing.T) {
-	err := featuregate.GlobalRegistry().Set("datadog.EnableOperationAndResourceNameV2", true)
-	if err != nil {
-		t.Fatal(err)
+func TestPushTraceDataOperationAndResourceName(t *testing.T) {
+	t.Run("OperationAndResourceNameV1", func(t *testing.T) {
+		testPushTraceDataOperationAndResourceName(t, false)
+	})
+
+	t.Run("OperationAndResourceNameV2", func(t *testing.T) {
+		testPushTraceDataOperationAndResourceName(t, true)
+	})
+}
+
+func testPushTraceDataOperationAndResourceName(t *testing.T, enableOperationAndResourceNameV2 bool) {
```

Comment:
nit: add `t.Helper()` both for cleaner stack traces as well as to signify that this is a helper. I would also change the name, it looks a bit too much like a test and not a helper with the current name (maybe `subtest`?)
```suggestion
func subtestPushTraceDataOperationAndResourceName(t *testing.T, enableOperationAndResourceNameV2 bool) {
	t.Helper()
```